src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2752,6 +2755,31 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
       ostree_kernel_args_append_argv (kargs, self->kernel_args_added);
     }
 
+  if (append_if_missing)
+    {
+      for (char **iter = append_if_missing; iter && *iter; iter++)
+        {
+          const char *arg = *iter;
+          if (!g_strv_contains (existing_kargs, arg))
```

Comment:
For a follow-up: there's a corner-case here: this doesn't account for `--append` itself already adding the karg. So e.g. `--append foo --append-if-missing foo` will append `foo` twice. (And similarly for `--delete`). Should be an easy fix, but we don't need to block on this!

Reply:
In that case, can we move the setting of `existing_kargs` just before `if (append_if_missing)` ? Or will that be a problem with the structure of the code? Other options but I think worse would be to refresh `existing_kargs` before the condition is checked or don't allow append and append-if-missing together.

Reply:
ok, sorry, I tested that and didn't work.

Reply:
Yeah, I think we can just lave it until we have https://github.com/ostreedev/ostree/issues/2329, which would fix this more easily.